win32ss/user/ntuser/cursoricon.c

```diff
@@ -134,52 +139,98 @@ IntRemoveCursorFromList(
 }
 
 VOID
-IntLoadSystenIcons(HICON hcur, DWORD id)
+IntLoadSystemCursors(HICON hcur, DWORD id)
 {
     PCURICON_OBJECT pcur;
     int i;
     PPROCESSINFO ppi;
 
-    if (hcur)
+    pcur = UserGetCurIconObject(hcur);
+    if (!pcur)
+    {
+        EngSetLastError(ERROR_INVALID_CURSOR_HANDLE);
+        return;
+    }
+
+    ppi = PsGetCurrentProcessWin32Process();
+
+    if (!(ppi->W32PF_flags & W32PF_CREATEDWINORDC))
+       return;
+
+    for (i = 0 ; i < 16; i++)
     {
-        pcur = UserGetCurIconObject(hcur);
-        if (!pcur)
+        if (gasyscur[i].type == id)
         {
-            EngSetLastError(ERROR_INVALID_CURSOR_HANDLE);
+            pcur->CURSORF_flags |= CURSORF_GLOBAL;
+            UserAssignmentLock((PVOID)&gasyscur[i].handle, pcur);
+            UserReferenceObject(pcur);
             return;
         }
+    }
+}
+
+VOID
+IntLoadSystemIcons(HICON hcur, DWORD id)
+{
+    PCURICON_OBJECT pcur;
+    int i;
+    PPROCESSINFO ppi;
+
+    pcur = UserGetCurIconObject(hcur);
+    if (!pcur)
+    {
+        EngSetLastError(ERROR_INVALID_CURSOR_HANDLE);
+        return;
+    }
 
-        ppi = PsGetCurrentProcessWin32Process();
+    ppi = PsGetCurrentProcessWin32Process();
 
-        if (!(ppi->W32PF_flags & W32PF_CREATEDWINORDC))
-           return;
+    if (!(ppi->W32PF_flags & W32PF_CREATEDWINORDC))
+       return;
 
-        // Set Small Window Icon and do not link.
-        if ( id == OIC_WINLOGO+1 )
+    // Set Small Window Icon and do not link.
+    if ( id == OIC_WINLOGO+1 )
+    {
+        pcur->CURSORF_flags |= CURSORF_GLOBAL;
+        UserReferenceObject(pcur);
+        gpsi->hIconSmWindows = UserHMGetHandle(pcur);
+        return;
+    }
+
+    for (i = 0 ; i < 6; i++)
+    {
+        if (gasysico[i].type == id)
         {
             pcur->CURSORF_flags |= CURSORF_GLOBAL;
+            UserAssignmentLock((PVOID)&gasysico[i].handle, pcur);
             UserReferenceObject(pcur);
-            pcur->head.ppi = NULL;
             return;
         }
+    }
+}
 
-        for (i = 0 ; i < 6; i++)
-        {
-            if (gasysico[i].type == id)
-            {
-                gasysico[i].handle = pcur;
-                pcur->CURSORF_flags |= CURSORF_GLOBAL;
+VOID
+IntUnloadSystemCursorsAndIcons(VOID)
+{
+    PCURICON_OBJECT pcur;
+    int i;
 
-                //
-                //  The active switch between LR shared and Global public.
-                //  This is hacked around to support this while at the initial system start up.
-                //
-                pcur->head.ppi = NULL;
+    /* Unload icons */
+    for (i = 0; i < 6; i++)
+    {
+        pcur = gasysico[i].handle;
+        if (!pcur) continue;
+        pcur->head.ppi = PsGetCurrentProcessWin32Process();
+        UserAssignmentUnlock((PVOID)&gasysico[i].handle);
+    }
 
-                IntInsertCursorIntoList(pcur);
-                return;
-            }
-        }
+    /* Unload cursors */
+    for (i = 0; i < 16; i++)
```

Comment:
magic number 16.

Reply:
Same answer as above. But different to above where you answered that question, we don't know what the 16 means here.